### PR TITLE
fix: use provider id from kubelet config file

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
The current `ProviderID` from the kubelet config file is not used
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
ProviderID field from kubelet config file is now working
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -487,7 +486,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		cadvisor:                                kubeDeps.CAdvisorInterface,
 		cloud:                                   kubeDeps.Cloud,
 		externalCloudProvider:                   cloudprovider.IsExternal(cloudProvider),
-		providerID:                              providerID,
+		providerID:                              kubeCfg.ProviderID,
```
Comment (andrewsykim) [inline pkg/kubelet/kubelet.go:489]: `providerID` is always unique per node. Typically you would generate the providerID dynamically by fetching it from a metadata server and then pass it into `--provider-id` flag. You would only do this if you wanted to override the providerID that would be set by the cloud provider. 

I'm not sure how valuable it is inherit this value from kubelet config cause kubelet config is global to all kubelets. If anything, we should maybe deprecate `providerID` in kubelet configuration.

Comment (Sh4d1) [reply pkg/kubelet/kubelet.go:489]: Hum 🤔 there is still one kubelet config file per node? Usually they are templated, but yeah I see your point.

**Comment by derekwaynecarr** [comment]
I agree with @andrewsykim that it may make more sense to just deprecate the option.

**Pull request closed without merging**